server/game_websocket/room.ts

```diff
@@ -1,71 +1,52 @@
 // this is where the rooms are managed and created
 import { v4 as uuidv4 } from "uuid";
-type Rooms = { users: number; id: string; board: Board };
+type Rooms = {
+  users: Array<{ username: string; color: string }>;
+  id: string;
+  board: Board;
+};
 import { Board } from "./board";
+import { RandomUser } from "./user";
 
 export class Room {
   // contains the state of the rooms (number of users, id)
-  private roomsDetails: Array<Rooms>;
+  private roomState: Array<Rooms>;
   // keeps tab of users added in new rooms
-  private roomsState: {
-    [key: string]: Array<{ username: string; color: string }>;
-  };
+  // private roomsState: {
+  //   [key: string]: Array<{ username: string; color: string }>;
+  // };
 
   private onlineUsers: number;
 
   constructor() {
     // TODO merge data structures into one optimal solution
     // no of users in room,
-    this.roomsDetails = [];
+    this.roomState = [];
     // list of actual users in a room
-    this.roomsState = {};
     this.onlineUsers = 0;
   }
 
-  public addUserToRoom(user: User) {
-    if (this.roomsState[user.roomID]) {
-      this.roomsState[user.roomID].push({
-        username: user.username,
-        color: user.color,
-      });
-    } else {
-      this.roomsState[user.roomID] = [
-        {
-          username: user.username,
-          color: user.color,
-        },
-      ];
-    }
-  }
-
-  public getUsersInRoom(user: User) {
-    return this.roomsState[user.roomID];
-  }
-
-  public removeUserFromRoom(user: User) {
-    this.roomsState[user.roomID] = this.roomsState[user.roomID].filter(
-      (usr) => usr.username != user.username
-    );
-  }
-
-  public joinRoom() {
+  public joinRoom(user: User) {
     return new Promise<{ id: string; board: Board }>((resolve) => {
-      for (let i = 0; i < this.roomsDetails.length; i++) {
-        if (this.roomsDetails[i].users < 3) {
-          this.roomsDetails[i].users++;
+      for (let i = 0; i < this.roomState.length; i++) {
+        if (this.roomState[i].users.length < 3) {
```

Comment:
`3` is magic number antipattern here. Lest create a variable with name ROOM_MAX_CAPACITY and use it instead.